workers/unity/Assets/Playground/Scripts/Worker/AndroidClientWorkerConnector.cs

```diff
@@ -0,0 +1,60 @@
+using Improbable.Gdk.Core;
+using Improbable.Gdk.Mobile;
+#if UNITY_ANDROID
+using Improbable.Gdk.Mobile.Android;
+#endif
+using System;
+using UnityEngine;
+
+
+namespace Playground
+{
+    public class AndroidClientWorkerConnector : MobileWorkerConnector, IMobileClient
+    {
+        public string IpAddress { get; set; }
+        public ConnectionScreenController ConnectionScreenController { get; set; }
+
+        [SerializeField] private GameObject level;
+
+        private GameObject levelInstance;
+
+        public async void TryConnect()
+        {
+            await Connect(WorkerUtils.AndroidClient, new ForwardingDispatcher()).ConfigureAwait(false);
+        }
+
+        protected override void HandleWorkerConnectionEstablished()
+        {
+            ConnectionScreenController.OnSuccess();
+
+            WorkerUtils.AddClientSystems(Worker.World);
+            if (level == null)
+            {
+                return;
+            }
+
+            levelInstance = Instantiate(level, transform);
+            levelInstance.transform.SetParent(null);
+        }
+
+        protected override void HandleWorkerConnectionFailure()
+        {
+            ConnectionScreenController.OnConnectionFailed();
+        }
+
+        protected override string GetHostIp()
+        {
+#if UNITY_ANDROID
+            if (Application.isMobilePlatform && DeviceInfo.IsAndroidStudioEmulator() && IpAddress.Equals(string.Empty))
+            {
+                return DeviceInfo.AndroidStudioEmulatorDefaultCallbackIp;
+            }
+
+            return IpAddress;
+#else
+            throw new PlatformNotSupportedException(
+                "This method is only defined for the Android platform. Please check your build settings.");
```

Comment:
Its not just this method - its the whole class really, right?

Reply:
Same comment in the iOS one

Reply:
This is literally only here, because the compiler is complaining. Being able to use this method on a different platform should be impossible, because of the assembly/platform limitations. An alternative is to wrap the entire class in preprocessor directives but I think it is much uglier+ it brings other problems of the class not being detected and what-not else.

Reply:
This was more a nit about the comment in the exception 😛